### PR TITLE
fix: Backticks get always escaped by runtime; add Ipython test

### PR DESCRIPTION
**Pull request by tobitege**
## Fixed Issue

The current runtime `run_ipython` always escapes all backticks in commands, regardless of the command being actual Python code or not. This breaks file operations in `agentskills` in that working with e.g. Typescript or Markdown files their output gets corrupted.
No attempt by an LLM to "fix" this was able to repair this as obviously the runtime wouldn't "allow" it.

I'm opening up this PR but with notice to not merge before I got more tests with OpenDevin and feedback from the maintainers.

### Previous PR

https://github.com/OpenDevin/OpenDevin/pull/2307

### Notes:
- `run_ipython` now does a preprocessing of the given command, so that backticks are only escaped if it determines that it is (valid) Python code
- added a new `test_ipython` with currently 2 tests to check that backticks are handled correctly
- in that script is a 3rd test for execution of multiple test commands, but have trouble getting it to work as it constantly has caused trouble, most like newlines

## Feedback wanted
I fear, that there are side effects to this approach, that I haven't thought of yet and I'm happy to get more advice on this topic.

cc @rbren @xingyaoww @li-boxuan

### CODE REVIEW

**Comment by iFurySt** [comment]
strange, this line should't be merged according to https://github.com/OpenDevin/OpenDevin/pull/2117#discussion_r1617737723 

```
action.code = action.code.replace('`', r'\`')
```

**Comment by tobitege** [comment]
> strange, this line should't be merged according to [#2117 (comment)](https://github.com/OpenDevin/OpenDevin/pull/2117#discussion_r1617737723)
> 
> ```
> action.code = action.code.replace('`', r'\`')
> ```

I see. Since your commit is in as well (with the EOL), we might just get rid of the replace completely.  
My assumption was that Python was the reason, not some weird SQL case.

File: tests/unit/test_ipython.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                call(expected_execute_command),
+            ]
+        )
+
+        assert (
+            test_code == result.content
+        ), f'The output should contain the expected print output, got: {result.content}'
+
+
+# DRAFT:
```
Comment (neubig) [inline tests/unit/test_ipython.py:77]: Looks good, but could you delete the dead code here?

Comment (tobitege) [reply tests/unit/test_ipython.py]: Done

**Review by neubig — APPROVED** [review]

**Pull request merged**